ra_aid/callbacks/default_callback_handler.py

```
@@ -259,46 +388,119 @@ def _extract_token_usage(self, response: LLMResult) -> dict:
+                limit_type, current_val, limit_val, exit_at_limit_config = limit_info
+                self._record_limit_reached(
+                    limit_type, current_val, limit_val, exit_at_limit_config
+                )
+
+                if self._cost_limit_user_decision_continue is True:
+                    # User previously approved, proceed.
+                    pass
+                elif self._cost_limit_user_decision_continue is False:
+                    # User previously disapproved, exit.
+                    print(
+                        f"Exiting due to {limit_type} limit ({current_val} >= {limit_val}) and previous user decision to not continue."
```

Pylint (3.3.7)

[convention] Line too long (135/100)

(C0301)